src/v/cluster/log_eviction_stm.cc

```diff
@@ -43,14 +43,57 @@ log_eviction_stm::log_eviction_stm(
   : base_t("log_eviction_stm.snapshot", logger, raft, kvstore) {}
 
 ss::future<> log_eviction_stm::start() {
-    ssx::spawn_with_gate(_gate, [this] { return monitor_log_eviction(); });
-    ssx::spawn_with_gate(
-      _gate, [this] { return handle_log_eviction_events(); });
+    setup_metrics();
+    // setup state watchdog timer
+    _state_watchdog_timer.set_callback([this] {
+        // When partition retention is disabled, the log eviction loop will be
+        // waiting for the events, we do not need to report that.
+        if (_loop_state.current_state == loop_state::waiting_for_events) {
+            return;
+        }
+        if (
+          _loop_state.last_update_timestamp
+            + config::shard_local_cfg()
+                .partition_manager_shutdown_watchdog_timeout()
+          < model::timeout_clock::now()) {
+            vlog(
+              _log.error,
+              "log eviction loop stopped making progress, current state: {}",
+              _loop_state);
+            _loop_state.stall_errors++;
+        }
+    });
+
+    ssx::spawn_with_gate(_gate, [this] {
+        return monitor_log_eviction().then([this] {
+            if (!_as.abort_requested()) {
+                vlog(
+                  _log.error,
+                  "log eviction loop exited unexpectedly, "
+                  "this should never happen. Current state: {}",
+                  _loop_state);
+            }
+        });
+    });
+    ssx::spawn_with_gate(_gate, [this] {
+        return handle_log_eviction_events().then([this] {
+            if (!_as.abort_requested()) {
+                vlog(
+                  _log.error,
+                  "log eviction loop exited unexpectedly, "
+                  "this should never happen. Current state: {}",
+                  _loop_state);
+            }
+            _loop_state.update_loop_state(loop_state::exited);
+        });
+    });
+    _state_watchdog_timer.arm_periodic(5s);
```

Comment:
Since this is going to be a timer that fires every 5s for each partition, I was wondering if we can cancel + create a timer /watchdog in  update_loop_state() (Since the whole timer machinery is optimized for creating + cancel pattern if not used rather than letting  then all fire at the same time, may have some unintended performance implications at scale)


```

update_loop_state() {
   _watch_dog.reset();

  // update state

   _watch_dog = WatchDog(5s)....

}

```

Probably optimized for the common case where there is no hang of the stm.